src/main/groovy/com/bmuschko/gradle/docker/DockerJavaApplicationPlugin.groovy

```diff
@@ -72,12 +72,15 @@ class DockerJavaApplicationPlugin implements Plugin<Project> {
         project.task(DOCKERFILE_TASK_NAME, type: Dockerfile) {
             description = 'Creates the Docker image for the Java application.'
             dependsOn tarTask
-
             from { dockerJavaApplication.baseImage }
             maintainer { dockerJavaApplication.maintainer }
             addFile({ tarTask.archivePath.name }, { '/' })
-            entryPoint { determineEntryPoint(project, tarTask) }
-            exposePort { dockerJavaApplication.getPorts() }
+            project.afterEvaluate {
+                // pass current delegate on to config closure
+                dockerJavaApplication.exec?.delegate = delegate
+                dockerJavaApplication.exec?.call()
+                exposePort { dockerJavaApplication.getPorts() }
```

Comment:
It was left inside to maintain the relative ordering but that may be shuffled around